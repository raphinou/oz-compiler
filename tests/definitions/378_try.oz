% Try with multiple catch clauses
local
   F R
in
   fun {F E}
      if {Record.width E}>0 then
      {Show raising#E}
         raise E end
      else
         raise error(E) end
      end
      ok
   end
  {Show try Res={F halt}
         in commit(Res)
         catch error(halt) then
            abort(halt completeMatch)
         [] error(E) then
            abort(E otherValueOfExc)
         [] failure(E) then
            {Show 'Failure'}
            E
        end }
  {Show try Res={F weeez}
         in commit(Res)
         catch error(halt) then
            abort(halt completeMatch)
         [] error(E) then
            abort(E otherValueOfExc)
         [] failure(E) then
            {Show 'Failure'}
            E
        end }
  {Show try Res={F failure(myError)}
         in commit(Res)
         catch error(halt) then
            abort(halt completeMatch)
         [] error(E) then
            abort(E otherValueOfExc)
         [] error2(E) then
            abort(E otherValueOfExc2)
         [] failure(E) then
            E
        end }
  {Show try Res={F failure(myError)}
         in commit(Res)
         catch error(halt) then
            abort(halt completeMatch)
         [] error(E) then
            abort(E otherValueOfExc)
         [] error2(E) then
            abort(E otherValueOfExc2)
         [] failure(E) then
            E
        end }
  {Show try Res={F failure(myError)}
         in commit(Res)
         catch error(halt) then
            abort(halt completeMatch)
         [] error(E) then
            abort(E otherValueOfExc)
         [] failure(E) then
            T
         in
            {Show 'Failure'}
            E
        end }
end
