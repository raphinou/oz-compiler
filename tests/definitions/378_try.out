abort(halt completeMatch)
abort(weeez otherValueOfExc)
raising#failure(myError)
myError
raising#failure(myError)
myError
raising#failure(myError)
'Failure'
myError
